src/raw.rs

```diff
@@ -491,9 +491,7 @@ where
     }
     macro_rules! f {
         () => {
-            CString::new((*fi.next().unwrap()).clone())
-                .unwrap()
-                .as_ptr()
+            fi.next().unwrap().as_ptr()
```

Comment:
Wouldn't we have the same problem as before? the `next` will return the CString which will be dropped right away and you will return its pointer. What am I missing?

Reply:
I do not think that the macro is the problem, working with pointer is unsafe and should be done carefully. Regardless, I do not like this macro as well ...

Reply:
We should probably add a proc macro on redismodule-rs-macros that will generated this call and this way we will also not be limited to the number of fields...

Reply:
I did some testing to check that the destructor is called at the correct time, but on second (third? fourth?) thought that testing might have been flawed and you might be right about the lifetime still being wrong.

Reply:
Not sure I understand how a proc macro would work, but it's been a while since I've dug into the details of Rust macros. I'm all in favor of it if it's possible.

Reply:
I suggest we will fix the current issue first, and open another issue to define this proc macro.